src/Psalm/Internal/Type/TypeParser.php

```diff
@@ -858,6 +863,10 @@ private static function getTypeFromGenericTree(
                 return new TPositiveInt();
             }
 
+            if (is_int($min_bound) && is_int($max_bound) && $min_bound > $max_bound) {
+                throw new TypeParseTreeException("Min bound can't be greater than max bound");
+            }
```

Comment:
[Wow](https://psalm.dev/r/4cdae92b1d), that check didn't exist before? :laughing:

I'm pretty sure the `CodeLocation` used for this error is going to be the entire type rather than just the int range, so this might be slightly confusing for something like `array<array-key, int<5, 3>>`, maybe mention `int<$min_bound, $max_bound>` in the message?

Reply:
Guilty as charged! (In fact, there's a few other places an IntegerRange could end up with inconsistent bounds due to bad checks). I briefly considered making TIntegerRange immutable in order to set up this check in constructor, but that was making a lot of code needlessly complex

Reply:
@orklah otoh if it was immutable [this](https://github.com/vimeo/psalm/issues/7697) difficult to debug issue would never have happened. I'd love for the type system to become immutable eventually, but I haven't really looked too hard at what it would take to make that happen.

Reply:
IIRC, the worst would be Union manipulation. You won't be able to change an Atomic state just by having the instance, you have to actually know its key and replace it inside the Union

Reply:
> this might be slightly confusing for something like array<array-key, int<5, 3>>, maybe mention int<$min_bound, $max_bound> in the message

You're right. Done!